### PR TITLE
improve .travis.yml

### PR DESCRIPTION
**Pull request by zserik**
I improved .travis.yml, so that it
* runs against nightly (to find regressions earlier) and
* compiles once with zcstream enabled and once without.

### CODE REVIEW

**Comment by zserik** [comment]
```
error[E0277]: the trait bound `tests::MockStream: zcstream::ZCStream` is not satisfied
   --> src/lib.rs:602:46
    |
602 |         let mut telnet = Telnet::from_stream(stream, 6);
    |                                              ^^^^^^ the trait `zcstream::ZCStream` is not implemented for `tests::MockStream`
    |
    = note: required for the cast to the object type `dyn zcstream::ZCStream`
```
ok, thats a regression, but I currently don't know how to fix it properly.

**Comment by SLMT** [comment]
Thanks. I will look into the problem.

**Comment by zserik** [comment]
You should probably take a look at the clippy output in the Travis CI build.

File: src/lib.rs [hunk 1]
```diff
@@ -595,6 +595,8 @@ mod tests {
         }
     }
 
+    // TODO: fix MockStream, so that it works with ZCStream
+    #[cfg(not(feature = "zcstream"))]
     #[test]
     fn escapes_double_iac_correctly() {
         let stream = Box::new(MockStream::new(vec!(0x40, 0x5a, 0xff, 0xff, 0x31, 0x34)));
```
Comment (SLMT) [inline src/lib.rs]: Try to change this line to the following code. It should be enough to solve the problem.

```rust
let stream = MockStream::new(vec!(0x40, 0x5a, 0xff, 0xff, 0x31, 0x34));
#[cfg(feature = "zcstream")]
let stream = ZlibStream::from_stream(stream);
let stream = Box::new(stream);
```

File: src/lib.rs [hunk 2]
```diff
@@ -547,6 +546,7 @@ impl Telnet {
 }
 
 #[cfg(test)]
+#[cfg_attr(feature = "zcstream", allow(dead_code, unused_imports))]
```
Comment (SLMT) [inline src/lib.rs]: Remember to remove this.

File: src/lib.rs [hunk 3]
```diff
@@ -595,6 +595,8 @@ mod tests {
         }
     }
 
+    // TODO: fix MockStream, so that it works with ZCStream
+    #[cfg(not(feature = "zcstream"))]
```
Comment (SLMT) [inline src/lib.rs]: Remember to remove this.

**Comment by zserik** [comment]
I probably rebase this branch again, that the "temporary disable of testing if feature=zcstream" doesn't show up in the final merge.

**Comment by zserik** [comment]
sorry, I forgot to update the commit message.

**Pull request merged**